src/datadog.ts

```diff
@@ -161,4 +164,9 @@ function validateProps(props: DatadogProps) {
       throw new Error("When `extensionLayer` is set, `apiKey` or `apiKmsKey` must also be set.");
     }
   }
+  if (props.architecture !== undefined) {
+    if (props.architecture !== "x86" && props.architecture !== "ARM") {
```

Comment:
Since CDK themselves use `ARM_64` and `X86_64`, I think we should use those as well to maintain consistency https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.AmazonLinuxCpuType.html